novikov_dmitry/logger.cpp

```diff
@@ -0,0 +1,27 @@
+#include "logger.hpp"
+#include <assert.h>
+#include <iostream>
+
+namespace uni_cpp_practice {
+
+void Logger::set_file(const std::optional<std::string>& filename) {
+  if (!filename.has_value()) {
+    file_stream_->close();
+    file_stream_ = std::nullopt;
+    return;
+  }
+  if (file_stream_->is_open()) {
+    file_stream_->close();
+  }
+
+  file_stream_ = std::ofstream(filename.value());
+  assert(file_stream_->is_open() && "Error while opening the log file!");

```

Comment:
Так как здесь ошибка вне нашего контроля, то это не проверка на баги, это исключительное состояние, лучше выкинуть `exception`
```suggestion
  if (!file_stream_->is_open()) {
    throw std::runtime_error("Failed to create file stream");
  }
```